packages/cli-repl/src/mongosh-repl.ts

```diff
@@ -391,6 +394,31 @@ class MongoshNodeRepl implements EvaluationListener {
     return { __initialized: 'yes' };
   }
 
+  setupHistoryCommand(): void {
+    const history = () => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const replHistory: string[] = (this.runtimeState().repl as any).history;
+      const formattedHistory =
+        // Remove the history call from the formatted history
+        replHistory.slice(1).reverse();
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      formattedHistory[util.inspect.custom as any] = (() => {
+        return formatOutput(
+          {
+            // Providing a copy of the history avoids a circular reference.
+            value: formattedHistory.concat(),
+          },
+          { colors: true, maxArrayLength: Infinity }
```

Comment:
Don't unconditionally set `colors: true` if you don't know if the output supports colors (or wants colors – disabling them can be an accessibility feature). We have `this.formatOutput()` on this class specifically to account for that.

All that being said, you'll likely want to follow best practices for Node.js custom inspect functions here and not use our `formatOutput()` at all and instead use the `inspectOptions` and `inspect` arguments that already get passed to the custom inspect function automatically.